### PR TITLE
Update CardBrowser.kt

### PR DESCRIPTION
**Pull request by zaidazaize**
## Fixes
- converted nullable to  lateinit var type
- Removed notnull (!!) assertions
- simplified comparison logic
- Edited UnitTests in CardBrowserTest file (removed not null (!!) assertions)

## How Has This Been Tested?
- Tests

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -187,8 +189,13 @@ open class CardBrowser :
             launchCatchingTask { saveEditedCard() }
         }
         val data = result.data
-        if (data != null &&
-            (data.getBooleanExtra("reloadRequired", false) || data.getBooleanExtra("noteChanged", false))
+        if ((data != null) &&
+            (
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:193]: Unnecessary indentation here

Comment (zaidazaize) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt]: done

Comment (zaidazaize) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt]: removed.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 2]
```diff
@@ -102,13 +102,14 @@ import kotlin.math.max
...
 @Suppress("LeakingThis") // The class is only 'open' due to testing
-@KotlinCleanup("scan through this class and add attributes - not started")
+@KotlinCleanup("scan through this class and add attributes - in process")
 open class CardBrowser :
     NavigationDrawerActivity(),
     SubtitleListener,
     DeckSelectionListener,
     TagsDialogListener,
     ChangeManager.Subscriber {
+
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:112]: Revert this unwanted line

**Review by criticalAY — APPROVED** [review]
Thankyou!
Approving, left a nit though

**Review by david-allison — CHANGES_REQUESTED** [review]
`onCollectionLoaded` is not guaranteed to be called.

The `lateinit` variables should be initialised in `onCreate` instead.

**Comment by zaidazaize** [comment]
cardsListView is now  initialised in onCreate

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 3]
```diff
@@ -132,21 +132,20 @@ open class CardBrowser :
...
-    @KotlinCleanup("move to onCreate and make lateinit")
     @VisibleForTesting
-    var cardsListView: ListView? = null
-    private var mSearchView: CardBrowserSearchView? = null
+    lateinit var cardsListView: ListView
+    private lateinit var mSearchView: CardBrowserSearchView
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    var cardsAdapter: MultiColumnListAdapter? = null
+    lateinit var cardsAdapter: MultiColumnListAdapter
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:140]: Same issue with `cardsAdapter`. Confirm that each variable is in `onCreate`

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 3 above)
Comment (zaidazaize) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:140]: moved initialization of `cardsAdapter` and `mActionBarTitle` to `onCreate`. Also have moved adapter linking from `onCollectionLoaded` to `onCreate`.

**Comment by david-allison** [comment]
Conflicts

**Pull request closed without merging**